### DISCUSSION TITLE
Element Index enhancements

### DISCUSSION THREAD

**Discussion by nateiler** [craftcms/cms#8685] [Ideas]
We're using element indexes in various ways.  However, there is a limitation with configuring the index settings which can only be assigned per Element Type.  I'm proposing to a 'context' column (defaulting to 'index') to 'elementindexsettings' table.  As a result, settings could be configured as various 'contexts'.

I've stepped through about 90% of what this would touch, and there are no breaking changes.  I'm happy to put together a pull request if it's worth entertaining.

**Comment by brandonkelly** [comment]
How would not-index context settings be managed? Would they default to the index settings?

**Comment by nateiler** [comment]
> How would not-index context settings be managed?

To not put a lot of burden on the core team nor introduce a lot of changes that isn't on the current roadmap, I would leave this up to third-party devs to implement if they so choose. 

> Would they default to the index settings?

Correct.  If a context doesn't exist, default to index.
